From 37bda348b9023b3ff445f1730e242ed917034a2a Mon Sep 17 00:00:00 2001
From: Active ESL <engineering@active-esl.com>
Date: Fri, 11 Sep 2026 16:00:00 +0100
Subject: [PATCH] platform: use Android 16 interface descriptor

LineageOS 23.2 exposes the Waydroid platform service as
id.waydro.waydroid.IPlatform. Older Android releases use the legacy
lineageos.waydroid.IPlatform descriptor.

Waydroid 1.6.3 already selects the AIDL6 service-manager protocol for API 36.
Use that selected protocol to choose the matching platform descriptor while
retaining compatibility with older images.
---
 tools/interfaces/IPlatform.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/tools/interfaces/IPlatform.py b/tools/interfaces/IPlatform.py
index b10e908..e0a41e8 100644
--- a/tools/interfaces/IPlatform.py
+++ b/tools/interfaces/IPlatform.py
@@ -7,6 +7,7 @@ import signal
 
 
 INTERFACE = "lineageos.waydroid.IPlatform"
+ANDROID_16_INTERFACE = "id.waydro.waydroid.IPlatform"
 SERVICE_NAME = "waydroidplatform"
 
 TRANSACTION_getprop = 1
@@ -24,7 +25,7 @@ TRANSACTION_launchIntent = 13
 
 class IPlatform:
-    def __init__(self, remote):
-        self.client = gbinder.Client(remote, INTERFACE)
+    def __init__(self, remote, interface=INTERFACE):
+        self.client = gbinder.Client(remote, interface)
 
     def getprop(self, arg1, arg2):
         request = self.client.new_request()
@@ -321,7 +322,9 @@ def get_service(args):
         else:
             return None
 
-    return IPlatform(remote)
+    interface = (ANDROID_16_INTERFACE if args.SERVICE_MANAGER_PROTOCOL == "aidl6"
+                 else INTERFACE)
+    return IPlatform(remote, interface)
 
 # Like ServiceManager.wait() but can be interrupted
 def wait_for_manager(sm):
-- 
2.43.0
